newIDE/app/src/EventsSheet/InstructionEditor/EventTextDialog.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+type RGBColor = {|
+  r?: number,
+  g?: number,
+  b?: number,
+  a?: number,
+|};
```

Comment:
Not too bad, but you probably don't want to mark r/g/b/a as optional. When doing this, then this means that you consider that these properties can be missing. It's a pain because that mean that you have to check everywhere if r/g/b and a are not undefined before using them! Instead, just say that a RGBColor is *always* containing these variables:

```
type RGBColor = {|
  r: number,
  g: number,
  b: number,
  a: number,
|};
```

Reply:
Also you might want to remove the "a", because you're not setting it anywhere (and Flow will warn you about not setting it if it's not optional).

Reply:
Actually, as you've asked:

> And i can't directly use type RGBColor ?

I guess you refer to the one in ColorPicker. You can use it if you:
* Add the "export" keyword before "type RGBColor" in ColorPicker
* Then import it: `import ColorPicker, { type RGBColor } from '../../UI/ColorField/ColorPicker';`

I recommend that you do this :) Better have a single place where a type is defined. This type has an extra optional `a` property, but that's fine!